### ISSUE TITLE
fix widget surveyor to work in switch statements

### ISSUE THREAD

**Issue by pq** [pq/surveyor#4]
To see where this is not working look at `accident_reporting`:

```dart
    MaterialApp(
      home: HomePage(),
      onGenerateRoute: (RouteSettings settings) {
        switch (settings.name) {
          case '/home':
            return PageTransition(child: HomePage(),type: PageTransitionType.leftToRight);
            break;
          case '/second':
            return PageTransition(child: PageOne(),type: PageTransitionType.leftToRight);
            break;
          case '/third':
            return PageTransition(
                child: PageTwo(
                  mapData: settings.arguments,
                ),type: PageTransitionType.leftToRight);
            break;
```

and notice that we do not track ~`PageTransition`s and~ `PageOne`, etc.

(**EDIT:** `PageTransition` is not a Widget...)

**Comment by pq** [comment]
Looking a snippet of the AST, the issue is that the `PageTransition` creations are being represented as `MethodInvocation`s (and not `InstanceCreationExpression`s as we're expecting).

```
SwitchCaseImpl [case '/home': return PageTransition(child: HomePage(), type: PageTransitionType.leftToRight); break;] 
                                SimpleStringLiteralImpl ['/home'] 
                                ReturnStatementImpl [return PageTransition(child: HomePage(), type: PageTransitionType.leftToRight);] 
                                  MethodInvocationImpl [PageTransition(child: HomePage(), type: PageTransitionType.leftToRight)] 
                                    SimpleIdentifierImpl [PageTransition] 
                                    ArgumentListImpl [(child: HomePage(), type: PageTransitionType.leftToRight)] 
                                      NamedExpressionImpl [child: HomePage()] 
                                        LabelImpl [child:] 
                                          SimpleIdentifierImpl [child] 
                                        MethodInvocationImpl [HomePage()] 
                                          SimpleIdentifierImpl [HomePage] 
                                          ArgumentListImpl [()] 
                                      NamedExpressionImpl [type: PageTransitionType.leftToRight] 
                                        LabelImpl [type:] 
                                          SimpleIdentifierImpl [type] 
                                        PrefixedIdentifierImpl [PageTransitionType.leftToRight] 
                                          SimpleIdentifierImpl [PageTransitionType] 
                                          SimpleIdentifierImpl [leftToRight] 
                                BreakStatementImpl [break;] 
```

@bwilkerson: is this surprising?  If it's helpful, the current implementation is in https://github.com/pq/surveyor/blob/master/example/widget_surveyor.dart.

**Comment by pq** [comment]
Fixed w/ be5b836a24bb0e67d2d6219cebce8e04cb463211.


```
[~/src/repos/surveyor] (master) $ cat accident_reporting_widget.csv
MaterialApp, 1, flutter
HomePage, 2, *
PageOne, 1, *
PageTwo, 1, *
PageThree, 1, *
PageFour, 1, *
PageFive, 1, *
PageSeven, 1, *
SwipeButtonDemo, 1, *
FullPageRadioGroup, 2, *
TopPrograssBar, 6, *
FullPageHumanAnatomy, 1, *
FullPageAnalogTimePicker, 1, *
FullPageLocationAndImagePicker, 1, *
DataReviewAndUpload, 1, *
```

/fyi @jayoung-lee

**Comment by jayoung-lee** [comment]
This is awesome! Thank you Phil.
